### ISSUE TITLE
Access Violation launching Notepad++ 8 with Dark Mode enabled in Windows

### ISSUE THREAD

**Issue by marekr** [notepad-plus-plus/notepad-plus-plus#9961]
### Description of the Issue
Updated to Notepad++ 8 64-bit

### Steps to Reproduce the Issue
1. Download installer
2.  Install Notepad++ 8
3. Run notepad++ while Windows 10 is in dark mode

### Expected Behavior
Runs just fine

### Actual Behavior
![image](https://user-images.githubusercontent.com/1838013/121039476-1d394a80-c77f-11eb-820a-9a4d97792c73.png)

### Debug Information
Windows Version	10.0.19043 Build 19043  **21H1**

**Turning off dark mode allows it to launch**

This may be incompatibility with 21H1 which was released a few weeks ago and Microsoft is now mass pushing to end users as of yesterday

**Comment by Hammerfest** [comment]
Fresh install of 21H1 a few days ago, and using the pre-release of 8 (never had v7 on it).
I don't have this issue.
I will download the full release and test this evening.

**Comment by marekr** [comment]
> Fresh install of 21H1 a few days ago, and using the pre-release of 8 (never had v7 on it).
> I don't have this issue.
> I will download the full release and test this evening.

I can build Notepad++ from source under VS2019 and proceed to run it under Dark mode with no issue.

However the release install fails to run still.

**Comment by Hammerfest** [comment]
> 
> 
> Fresh install of 21H1 a few days ago, and using the pre-release of 8 (never had v7 on it).
> I don't have this issue.
> I will download the full release and test this evening.

To correct this, the pre-release v8 was done on an old windows 10 install (20H2).
On my desktop I am on now which is the new fresh Win 10 install (21H1) and has never had NP++ on it, fresh install from the download page just now for x64 installer worked fine and starts with no issues.

**Comment by marekr** [comment]
Are there PDBs sitting around for use with released builds? It would make identifying the issue easier :/

![image](https://user-images.githubusercontent.com/1838013/121076144-12dd7780-c7a4-11eb-8c9e-c51a05522c83.png)

**Comment by xcasxcursex** [comment]
FYI I'm using 21H1 and dark mode is working here. 
![image](https://user-images.githubusercontent.com/58130821/121123690-fce4bb00-c866-11eb-9c6e-179944bd00f7.png)

**Comment by xcasxcursex** [comment]
FYI I'm using 21H1 and dark mode is working fine here.

**Comment by gerard-durand** [comment]
Tried this V8 version, installed over a 7.95 on a 21H1 windows recently updated. I have not enabled dark theme on windows, but I obtained the same behavior as described in the first post when I simply open a txt file. After a while and several tries, the same txt file opens correctly, but after closing it, I can't again open it (same error). So, I switched back by reinstalling 7.95 which works as previously.

**Comment by donho** [comment]
@gerard-durand  @marekr 
Could you provide your debug Info please? (Menu `?->Debug Info`)

**Comment by donho** [comment]
@marekr 
> Are there PDBs sitting around for use with released builds? It would make identifying the issue easier :/

PDB 32 bits:
[npp.pdb.32.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/6620478/npp.pdb.32.zip)

PDB 64 bits:
[npp.pdb.64.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/6620488/npp.pdb.64.zip)

**Comment by marekr** [comment]
> @gerard-durand @marekr
> Could you provide your debug Info please? (Menu `?->Debug Info`)

Nothing exciting there, it's literally the released v8 install

```
Notepad++ v8   (64-bit)
Build time : Jun  7 2021 - 13:30:08
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit) 
OS Version : 2009
OS Build : 19043.985
Current ANSI codepage : 1252
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExport.dll 
```

**Comment by marekr** [comment]
> @marekr
> 
> > Are there PDBs sitting around for use with released builds? It would make identifying the issue easier :/
> 
> PDB 32 bits:
> [npp.pdb.32.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/6620478/npp.pdb.32.zip)
> 
> PDB 64 bits:
> [npp.pdb.64.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/6620488/npp.pdb.64.zip)

![image](https://user-images.githubusercontent.com/1838013/121354445-e25d2100-c8fc-11eb-8053-b1632b9af6c7.png)

**Comment by marekr** [comment]
It appears it's do to the BITMAP_INFO instance containing a ridiculous height

![image](https://user-images.githubusercontent.com/1838013/121354967-66afa400-c8fd-11eb-9476-ee57a7aafff5.png)

**Comment by donho** [comment]
@marekr 
Thank you for the information!

@adzm 
Do you have any idea of the value of biHeight?

**Comment by marekr** [comment]
Fyi I looked at it further and it might be a race condition where this code path to setup dark mode icons is being run based on events before the toolbar icons got constructed. Hard to tell without debugging more closely.

**Comment by donho** [comment]
@marekr Thank you for your information.
I tried to do some fixes, not in the direction of race condition but only check the result of a function call.
Could you please try it out and tell me what it's happening?
http://download.notepad-plus-plus.org/repository/MISC/npp64.zip

**Comment by marekr** [comment]
> @marekr Thank you for your information.
> I tried to do some fixes, not in the direction of race condition but only check the result of a function call.
> Could you please try it out and tell me what it's happening?
> http://download.notepad-plus-plus.org/repository/MISC/npp64.zip

It seems to startup just fine now, I haven't encountered a crash yet.

However, I encountered a few times a crash going into Preferences and turning off Dark Mode. Only 4 times, the 5th time it worked without issue.

**Comment by donho** [comment]
@marekr 
Thanks to your information I guess I've fixed part of crashes. Feel free to provide any info here if you can reproduce the crash again.

**Comment by gerard-durand** [comment]
Hi,
I tried also this new version. Apparently, no problems after many tries. I cross my fingers ! Thanks.

**Comment by donho** [comment]
@marekr 
No more crash so far?

**Comment by donho** [comment]
@marekr 
> The dark mode preference toggle crash

Doesn't the provided PDB work for debugging?

**Comment by donho** [comment]
@marekr @gerard-durand 
I just modified the code concerning the crash you've got (that I fixed) for fixing a regression.
After fixing the regression, it could make the crash back.
Could you try the binaries and confirm me that the crash does not happen in v8.1 RC 2 please?
https://community.notepad-plus-plus.org/topic/21337/notepad-v8-1-release-candidate-2

**Comment by marekr** [comment]
I can't get it to crash on startup anymore that is fine.

Reproducing the crash on the preferences checkbox is hard. It only happens over remote desktop which I believe is because the computer I'm RDPing too gets slowed down alot doing so. The slow down is most likely delaying and reordering window events.

However, WinDbg seems to affect the priority of the process and I haven't been able to catch the crash.

**Comment by gerard-durand** [comment]
Tested the 8.1 RC2 version on my Windows 10 64 bits. No more problems apparently after testing it on several files.

**Comment by donho** [comment]
I consider the crash issue has been fixed in v8.1.
